target_redshift/db_sync.py

```diff
@@ -488,66 +492,153 @@ def load_csv(self, s3_key, count, size_bytes, compression=False):
-                    cur.execute(insert_sql)
-                    inserts = cur.rowcount
-
-                # Step 5/b: Insert only if no primary key
-                else:
-                    insert_sql = """INSERT INTO {} ({})
-                        SELECT {}
-                        FROM {} s
-                    """.format(
-                        target_table,
-                        ', '.join([c['name'] for c in columns_with_trans]),
-                        ', '.join(['s.{}'.format(c['name']) for c in columns_with_trans]),
-                        stage_table
-                    )
-                    self.logger.debug("Running query: {}".format(insert_sql))
-                    cur.execute(insert_sql)
-                    inserts = cur.rowcount
+                        self.logger.debug("Running query: {}".format(insert_sql))
+                        cur.execute(insert_sql)
+                        inserts = cur.rowcount
+
+                    # Step 5/b: Insert only if no primary key
+                    else:
+                        insert_sql = """INSERT INTO {} ({})
+                            SELECT {}
+                            FROM {} s
+                        """.format(
+                            target_table,
+                            ', '.join([c['name'] for c in columns_with_trans]),
+                            ', '.join(['s.{}'.format(c['name']) for c in columns_with_trans]),
+                            stage_table
+                        )
+                        self.logger.debug("Running query: {}".format(insert_sql))
+                        cur.execute(insert_sql)
+                        inserts = cur.rowcount
+
+                try:
+                    load_data()
+                except InternalError_ as exc:
+                    if (
+                        "Value too long for character type" in repr(exc)
+                        and not self.column_size_increase_attempted
+                    ):
+                        self.logger.error("Value too long for character type.")
+                        self.logger.error(
+                            "Will try to increase the size of all varchar columns"
+                        )
+                        try:
+                            self._increase_all_varchar_columns_to_max_size(
+                                target_table, self.schema_name
+                            )
+                        except Exception as exc:
+                            self.logger.error(
+                                f"Error running columns size increase: {exc}"
+                            )
+                        finally:
+                            self.column_size_increase_attempted = True
+
+                        self.logger.info(
+                            "Trying again after column size increase attempt"
+                        )
+                        load_data()                
```

Comment:
So if it fails twice we'll let it raise the exception, right?

Reply:
yeap - it was swallowing the exception, thanks for catching!